### PR TITLE
test: add integration tests for approve borrow request

### LINKED ISSUE #195: ApproveBorrowRequest tests

### PR DESCRIPTION
**Pull request by StarryFolf**

### CODE REVIEW

File: tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        // Act
+        var result = async () => await SendAsync(command);
+
+        // Assert
+        await result.Should().ThrowAsync<KeyNotFoundException>()
+            .WithMessage("Borrow request does not exist.");
+    }
+
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenDocumentIsLost()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs:68]: I would suggest asserting for if the result request is in NotProcessable state

File: tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        var document = CreateNDocuments(1).First();
+
+        var user1 = CreateUser(IdentityData.Roles.Employee, "abcdef");
+        var user2 = CreateUser(IdentityData.Roles.Employee, "aaaaaa");
+
+        var request1 = CreateBorrowRequest(user1, document, BorrowRequestStatus.Approved);
+        
+        var request2 = CreateBorrowRequest(user2, document, BorrowRequestStatus.Pending);
+        request2.BorrowTime = request2.BorrowTime.Plus(Period.FromHours(1));
+        request2.BorrowTime = request2.BorrowTime.Plus(Period.FromHours(1));
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs]: you forgot to change to request1

**Review by ChienNQuang — CHANGES_REQUESTED** [review]
A few things need to be adjusted

File: tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs (hunk 2 above)
Comment (StarryFolf) [reply tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs]: clarification, the second line should have changed the due time of the request, but i forgot

File: tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs (hunk 1 above)
Comment (StarryFolf) [reply tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs:68]: how would you assert the result if it throws an exception before returning anything?

File: tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs (hunk 2 above)
Comment (ChienNQuang) [reply tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs]: reasonable

**Review by ChienNQuang — APPROVED** [review]
LGTM

**Pull request merged**